File: backend/src/main/java/com/gymapp/service/WorkoutSaveService.java
package com.gymapp.service;

import com.gymapp.dto.request.WorkoutSaveRequest;
import com.gymapp.dto.response.WorkoutSaveResponse;

import java.util.List;

public interface WorkoutSaveService {
    List<WorkoutSaveResponse> getAllWorkoutSaves();

    WorkoutSaveResponse getWorkoutSaveById(Long id);

    List<WorkoutSaveResponse> getWorkoutSavesByWorkout(Long workoutId);

    List<WorkoutSaveResponse> getWorkoutSavesByUser(Long userId);

    WorkoutSaveResponse createWorkoutSave(WorkoutSaveRequest request);

    WorkoutSaveResponse updateWorkoutSave(Long id, WorkoutSaveRequest request);

    void deleteWorkoutSave(Long id);
}

File: gymapp/src/services/workoutSaveService.js
import { apiRequest } from "./apiClient";

const API_URL = "/api/workout-saves";

export async function getAllWorkoutSaves() {
  return apiRequest(API_URL);
}

export async function getWorkoutSaveById(id) {
  return apiRequest(`${API_URL}/${id}`);
}

export async function getWorkoutSavesByWorkout(workoutId) {
  return apiRequest(`${API_URL}/workout/${workoutId}`);
}

export async function getWorkoutSavesByUser(userId) {
  return apiRequest(`${API_URL}/user/${userId}`);
}

export async function createWorkoutSave(workoutSaveData) {
  return apiRequest(API_URL, {
    method: "POST",
    headers: {
      "Content-Type": "application/json",
    },
    body: JSON.stringify(workoutSaveData),
  });
}

export async function updateWorkoutSave(id, workoutSaveData) {
  return apiRequest(`${API_URL}/${id}`, {
    method: "PUT",
    headers: {
      "Content-Type": "application/json",
    },
    body: JSON.stringify(workoutSaveData),
  });
}

export async function deleteWorkoutSave(id) {
  await apiRequest(`${API_URL}/${id}`, {
    method: "DELETE",
  });
}
